wordpress-mobile/WordPress-iOS
# [Jetpack] Shared Defaults Utility Variables

Fixes #19235 

Extracts the utility variables to a protocol extension `UserPersistentRepositoryUtility` instead of extending the `UserDefaults` directly. The only remaining user default access' are from the file called `CommentsViewController`. Although as this is an objective-c file, it will require a bunch of modules to be marked as `@objc`. The 2 defaults don't seem critical to me and I rather keep it unpolluted for just the two.

Feature toggle is still set to false. It will be enabled once we have green light for release.

To test:
1. Install & Log in WP.
2. Set Shared Defaults Flag to true (so that both apps run on true)
3. Modify a persisting user setting like App Theme.
4. Run JP and Log in.
5. Enable the feature flag in JP. (You might need to re-run the JP app after if the default you're checking is only read in launch)
6. Check if changes reflect there.

## Regression Notes
1. Potential unintended areas of impact
Even though the change is repetitive, it is pretty all around. A lot of files are refactored. There were `value` calls instead of `object` which is (the former) not a `UserDefaults` function. The intended way is using `object` so I refactored accordingly.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
We have to manually test this. Ideally we enable the toggle after a release to have 2 weeks to test regression.

3. What automated tests I added (or what prevented me from doing so)
Changes are quite spread on legacy code. Testing it would require multitudes of more refactoring.

PR submission checklist:
